src/bun.js/bindings/NodeVM.cpp

```diff
@@ -738,6 +738,19 @@ Structure* NodeVMGlobalObject::createStructure(JSC::VM& vm, JSC::JSValue prototy
 
 void unsafeEvalNoop(JSGlobalObject*, const WTF::String&) {}
 
+static void promiseRejectionTrackerForNodeVM(JSGlobalObject* globalObject, JSC::JSPromise* promise, JSC::JSPromiseRejectionOperation operation)
+{
+    // JSInternalPromise should not be tracked through the normal promise rejection mechanism
+    // as they are internal to the engine and should not be exposed to user space.
+    if (jsDynamicCast<JSC::JSInternalPromise*>(promise))
```

Comment:
I'm skeptical this will not lead to bugs. Was there a test failure this fixed? If there was no test failure, can we remove it and then do it as a standalone PR in a follow-up that adds a test for promise rejections in various scenarios